examples/mnist.py

```diff
@@ -100,29 +110,35 @@ def create_iter_functions(dataset, output_layer,
                           X_tensor_type=T.matrix,
                           batch_size=BATCH_SIZE,
                           learning_rate=LEARNING_RATE, momentum=MOMENTUM):
+    """Create functions for training, validation and testing to iterate one
+       epoch.
+    """
     batch_index = T.iscalar('batch_index')
     X_batch = X_tensor_type('x')
     y_batch = T.ivector('y')
-    batch_slice = slice(
-        batch_index * batch_size, (batch_index + 1) * batch_size)
+    batch_slice = slice(batch_index * batch_size,
+                        (batch_index + 1) * batch_size)
 
     objective = lasagne.objectives.Objective(output_layer,
         loss_function=lasagne.objectives.categorical_crossentropy)
 
-    loss_train = objective.get_loss(X_batch, target=y_batch)
-    loss_eval = objective.get_loss(X_batch, target=y_batch,
-                                   deterministic=True)
+    loss = {'train': objective.get_loss(X_batch, target=y_batch),
+            'eval': objective.get_loss(X_batch,
+                                       target=y_batch,
+                                       deterministic=True)
+           }
```

Comment:
I think it would be better to keep it as is, as you said it's not really possible to easily 'generalize' the creation of theano functions across these two cases, so using a dictionary here seems superfluous and it just makes the code a tiny bit harder to understand.